### PR TITLE
AESH Upgrade and associated CLI refactoring

### PR DESCRIPTION
**Pull request by Gonthim**

### CODE REVIEW

**Comment by Gonthim** [comment]
Most of these errors don't make sense. The line described, CLITestUtil 70, was changed. There shouldn't be a reference to the old AESH Settings.getInstance() anymore.

Somehow the wildfly-integration test run seems to be getting the new AESH 0.62, which doesn't contain the Settings class anymore, but is still using the current CLITestUtil rather than the new one in the commit.

**Comment by bstansberry** [comment]
The testsuite/shared/pom.xml in full is directly declaring the org.wildfly.core:wildfly-core-testsuite-shared version instead of relying on dependency-management in the parent:

```
            <!-- Not defined in the parent pom as this should only be used here -->
            <dependency>
                <groupId>org.wildfly.core</groupId>
                <artifactId>wildfly-core-testsuite-shared</artifactId>
                <version>${version.org.wildfly.core}</version>
            </dependency>
```

I don't know why that would fail though. The -Dversion.org.wildfly.core=2.0.0.Alpha11-SNAPSHOT passed to maven should still be resulting in the correct version being used.

File: pom.xml [hunk 1]
```diff
@@ -705,6 +705,12 @@
 
             <dependency>
                 <groupId>org.wildfly.core</groupId>
+                <artifactId>wildfly-core-testsuite-shared</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
```
Comment (Gonthim) [inline pom.xml]: I'm not sure if adding the testsuite-shared dependency to the dependencyManagement in the root pom is an acceptable way to solve this problem, but it seemed to work locally. If it passes at least it gives us a clue to the cause.

**Comment by bstansberry** [comment]
retest this please

File: pom.xml (hunk 1 above)
Comment (bstansberry) [inline pom.xml]: I don't see any reason what that module shouldn't be declared there.

Comment (bstansberry) [inline pom.xml]: OT: this is just my personal opinion and isn't any sort of policy, but I quite dislike org.wildfly.core:wildfly-core-testsuite-shared being used outside of WildFly Core. I don't think people working on a testsuite utility module will put the kind of care into it that's appropriate for public API used by other projects.

**Comment by Gonthim** [comment]
Well, the one error I can replicate locally is the CLISecurityTestCase issue. Not with the error that shows up here, frustratingly, but with something similar so I think it's related.

Happily, I can adjust that test to pass both with and without the upgrade, so fixing it doesn't cause too much of a problem.

Related PR: https://github.com/wildfly/wildfly/pull/7823

**Comment by Gonthim** [comment]
retest this please

**Comment by bstansberry** [comment]
http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=62070&tab=buildResultsDiv&buildTypeId=WF_WildFlyCoreIntegrationExperiments is a custom test of the combination of this branch and the branch used for https://github.com/wildfly/wildfly/pull/7823. 

@Gonthim please ping me in hipchat if you need to learn how to kick off a CI job like this.

**Comment by bstansberry** [comment]
retest this please

http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=62070&tab=buildResultsDiv&buildTypeId=WF_WildFlyCoreIntegrationExperiments looked good and I've merged https://github.com/wildfly/wildfly/pull/7823

**Comment by bstansberry** [comment]
It's seems the pull request jobs are not setting -Dts.patching so that module doesn't even get built. When it does, this pops up:

```

[21:00:53][Step 2/3] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:testCompile (default-testCompile) on project wildfly-core-ts-patching: Compilation failure: Compilation failure:
[21:00:53][Step 2/3] [ERROR] /opt/buildAgent/work/47f11bb62c11a766/testsuite/patching/src/test/java/org/jboss/as/test/patching/PatchStreamsUnitTestCase.java:[100,49] cannot find symbol
[21:00:53][Step 2/3] [ERROR] symbol:   method getInstance()
[21:00:53][Step 2/3] [ERROR] location: interface org.jboss.aesh.console.settings.Settings
[21:00:53][Step 2/3] [ERROR] /opt/buildAgent/work/47f11bb62c11a766/testsuite/patching/src/test/java/org/jboss/as/test/patching/PatchInfoTestCase.java:[225,49] cannot find symbol
[21:00:53][Step 2/3] [ERROR] symbol:   method getInstance()
[21:00:53][Step 2/3] [ERROR] location: interface org.jboss.aesh.console.settings.Settings
[21:00:53][Step 2/3] [ERROR] -> [Help 1]
[21:00:53][Step 2/3] [ERROR] 
[21:00:53][Step 2/3] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[21:00:53][Step 2/3] [ERROR] Re-run Maven using the -X switch to enable full debug logging.
[21:00:53][Step 2/3] [ERROR] 
[21:00:53][Step 2/3] [ERROR] For more information about the errors and possible solutions, please read the following articles:
[21:00:53][Step 2/3] [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[21:00:53][Step 2/3] [ERROR] 
[21:00:53][Step 2/3] [ERROR] After correcting the problems, you can resume the build with the command
[21:00:53][Step 2/3] [ERROR]   mvn <goals> -rf :wildfly-core-ts-patching
[21:00:53][Step 2/3] Process exited with code 1
```

**Comment by Gonthim** [comment]
retest this please

**Comment by Gonthim** [comment]
These seem unrelated. The 4 similar failures from the last 2 attempts also failed in 3 other recent PR builds. Can't replicate them locally with or without this commit.

**Comment by bstansberry** [comment]
The last 4 failures were due to some process being left around on the CI agent. The 2nd one Jeff Mesnil is about to fix (unrelated). The first is a known intermittent failure.

**Comment by bstansberry** [comment]
@Gonthim FYI: https://issues.jboss.org/browse/WFCORE-841

**Pull request merged**